### PR TITLE
kamilalisp: init at 0.2

### PR DESCRIPTION
**Pull request by cafkafk**
A functional, flexible and concise Lisp inspired by Haskell and APL, among others. A ground-up rewrite.

Release notes:
https://github.com/kspalaiologos/kamilalisp/releases/tag/v0.2o

###### Description of changes
Added the program via a jar wrapper, as advised by the program author.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/interpreters/kamilalisp/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchurl, jre, lib, makeWrapper }:
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/kamilalisp/default.nix:1]: One per line, for better diffs

```suggestion
{ lib
, stdenv
, fetchurl
, jre
, makeWrapper
}:
```

File: pkgs/development/interpreters/kamilalisp/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+stdenv.mkDerivation rec {
+  name = "kamilalisp";
+  version = "0.2";
+
+  src = fetchurl {
+    url = "https://github.com/kspalaiologos/kamilalisp/releases/download/v0.2o/kamilalisp-0.2.jar";
+    sha256 = "sha256-NkRkLcJpar9iHEI+OmgpkYwkwVvU1J7izeBdytwIsdU=";
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/kamilalisp/default.nix:9]:
```suggestion
    url = "https://github.com/kspalaiologos/kamilalisp/releases/download/v${version}o/kamilalisp-${version}.jar";
    hash = "sha256-NkRkLcJpar9iHEI+OmgpkYwkwVvU1J7izeBdytwIsdU=";
```

File: pkgs/development/interpreters/kamilalisp/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  };
+
+  dontUnpack = true;
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
+  mkdir -pv $out/share/java $out/bin
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/kamilalisp/default.nix:17]: Indent.

File: pkgs/development/interpreters/kamilalisp/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    --set _JAVA_AWT_WM_NONREPARENTING 1
+  '';
+
+  meta = {
+    homepage = "https://github.com/kspalaiologos/kamilalisp.git";
+    description = "A functional, flexible, and concise Lisp.";
+    license = lib.licenses.gpl3Plus;
+    platforms = lib.platforms.all;
+    maintainers = with lib.maintainers;
+      [ cafkafk ];
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/kamilalisp/default.nix:31]:
```suggestion
    maintainers = with lib.maintainers; [ cafkafk ];
```

File: pkgs/development/interpreters/kamilalisp/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  mkdir -pv $out/share/java $out/bin
+  cp ${src} $out/share/java/${name}-${version}.jar
+  makeWrapper ${jre}/bin/java $out/bin/kamilalisp \
+    --add-flags "-jar $out/share/java/${name}-${version}.jar" \
+    --set _JAVA_OPTIONS '-Dawt.useSystemAAFontSettings=on' \
+    --set _JAVA_AWT_WM_NONREPARENTING 1
+  '';
+
+  meta = {
+    homepage = "https://github.com/kspalaiologos/kamilalisp.git";
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/kamilalisp/default.nix:26]:
```suggestion
    homepage = "https://github.com/kspalaiologos/kamilalisp";
```

File: pkgs/development/interpreters/kamilalisp/default.nix [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    description = "A functional, flexible, and concise Lisp.";
+    license = lib.licenses.gpl3Plus;
+    platforms = lib.platforms.all;
+    maintainers = with lib.maintainers;
+      [ cafkafk ];
+  };
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/kamilalisp/default.nix:32]:
```suggestion
    sourceProvenance = [ lib.sourceTypes.binaryBytecode ];
  };
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Can't this be built from sources?

**Comment by cafkafk** [comment]
> Can't this be built from sources?

It uses maven, as far as I could tell that is very non trivial to build with nix. I couldn't find any examples of it being done in nixpkgs, and my attempts getting it to work all failed. 

Also, the build time was, in my limited testing (not with nix), very long.

File: pkgs/development/interpreters/kamilalisp/default.nix (hunk 3 above)
Comment (cafkafk) [reply pkgs/development/interpreters/kamilalisp/default.nix:17]:
```nix
  installPhase = ''
    mkdir -pv $out/share/java $out/bin
    cp ${src} $out/share/java/${name}-${version}.jar
    makeWrapper ${jre}/bin/java $out/bin/kamilalisp \
      --add-flags "-jar $out/share/java/${name}-${version}.jar" \
      --set _JAVA_OPTIONS '-Dawt.useSystemAAFontSettings=on' \
      --set _JAVA_AWT_WM_NONREPARENTING 1
  '';
```
you mean this part right?

File: pkgs/development/interpreters/kamilalisp/default.nix (hunk 6 above)
Comment (cafkafk) [reply pkgs/development/interpreters/kamilalisp/default.nix:32]: ohh didn't know about this, that's neat!

File: pkgs/development/interpreters/kamilalisp/default.nix [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    cp ${src} $out/share/java/${name}-${version}.jar
+    makeWrapper ${jre}/bin/java $out/bin/kamilalisp \
+      --add-flags "-jar $out/share/java/${name}-${version}.jar" \
+      --set _JAVA_OPTIONS '-Dawt.useSystemAAFontSettings=on' \
+      --set _JAVA_AWT_WM_NONREPARENTING 1
+  '';
+
+  meta = {
+    homepage = "https://github.com/kspalaiologos/kamilalisp";
+    description = "A functional, flexible, and concise Lisp.";
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/kamilalisp/default.nix:32]:
```suggestion
    description = "A functional, flexible, and concise Lisp";
```

File: pkgs/development/interpreters/kamilalisp/default.nix [hunk 8]
```diff
@@ -0,0 +1,38 @@
+{ lib,
+  stdenv,
+  fetchurl,
+  jre,
+  makeWrapper
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/kamilalisp/default.nix:5]:
```suggestion
{ lib
, stdenv
, fetchurl
, jre
, makeWrapper
```

File: pkgs/development/interpreters/kamilalisp/default.nix [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+      --add-flags "-jar $out/share/java/${name}-${version}.jar" \
+      --set _JAVA_OPTIONS '-Dawt.useSystemAAFontSettings=on' \
+      --set _JAVA_AWT_WM_NONREPARENTING 1
+  '';
+
+  meta = {
+    homepage = "https://github.com/kspalaiologos/kamilalisp";
+    description = "A functional, flexible, and concise Lisp.";
+    license = lib.licenses.gpl3Plus;
+    platforms = lib.platforms.all;
```
Comment (AndersonTorres) [inline pkgs/development/interpreters/kamilalisp/default.nix:34]:
```suggestion
    inherit (jre.meta) platforms;
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Pull request merged**